src/test/kotlin/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTServiceTest.kt

```diff
@@ -56,39 +68,44 @@ class InformatieObjectenRESTServiceTest : BehaviorSpec() {
     @InjectMockKs
     lateinit var informatieObjectenRESTService: InformatieObjectenRESTService
 
-    override suspend fun beforeTest(testCase: TestCase) {
+    override suspend fun beforeContainer(testCase: TestCase) {
```

Comment:
Nice that you figured this out! Just checking: we really need to use `beforeContainer` here? Seems like such a big scope.

Reply:
Container is `Given` and `When` and for this particular test we want mocks to be cleaned before Given, so seemed appropriate.

We can also do with `beforeTest` but it called 7 more times (32 vs 39)

Reply:
Sorry @hsiliev I really don't know. I never heard of the container lifecycle before. It just seems that it is meant for stuff that spans multiple test cases and not stuff within one test case which is our context right?
https://kotest.io/docs/framework/lifecycle-hooks.html

Reply:
Does your solution also work when we apply it to all our other test cases?

Reply:
Container in kotest seems to be `describe`, `context`, `given`, `when`, `then`. I think everything that contains tests inside.

What we have in this PR is specific to given-when-then with setup done in given. 

We can increase complexity with nested `before*` but it becomes hard to grasp and hard to maintain. Currently the file is already asking for split into multiple files and we haven't covered even 1/10 of the impl

If we stick to simple tests/setup we should be good-to-go for the rest of the tests too. On the other hand I'm sure there will be exceptions :)

Reply:
ok thanks for the explanation!